### ISSUE TITLE
Allow the Digital Storage Unit to Act as a Crafting Grid Instead of a Grid

### ISSUE THREAD

**Issue by RobuxShooters** [MorningSage/Mekanism_Addons#16]
**[For Refined Storage]**
With a second Digital Storage Unit installed in the Meka-Tool, the Meka-Tool should be able to act as a Wireless Crafting Grid instead of a normal Crafting Grid, since a Wireless Crafting Grid is definitely more useful in terms of functionality as opposed to a normal Grid which does not have any crafting capabilities. The appeal, for the most part, is the ability to quickly craft things in the Wireless Grid without having to use an external workbench/crafting table and for JEI to work with the assortment of items in the Refined Storage system without needing to individually take things out of the system just to craft things.

This should not phase out or displace the functionality of the Wireless Crafting Grid either, as this option is definitely more expensive compared to the Wireless Crafting Grid from the base mod itself (Refined Storage Addons), which is much cheaper but takes up an inventory slot as opposed to this unit.

**Comment by MorningSage** [comment]
The ultimate goal is to have both AE2 and RS both have a crafting terminal/grid.  However, to avoid having a large code base (and because I’m lazy), I’m actually using the terminal/grid from the mods themselves and opening the the respective mod’s portable screen (RS just doesn’t have a portable crafting grid in the base mod like AE2 does).

I’d love to use the best grid available and possibly to tie into some addons for those mods that add others (like ae2wtlib, or whatever it is, and the RS one mentioned by the person who made this suggestion originally).

I’m on holiday through the rest of the year without my computer. As soon as I’m back, I’ll look into making this change either by manually adding a custom terminal/grid or by using a crafting screen from both mods.

Thanks and sorry for the delay.

**Comment by RobuxShooters** [comment]
That is true, RS strangely doesn't have a Wireless Crafting Grid in the base mod but includes it in an addon (Refined Storage Addons). Would it be possible to reuse existing code from the addon mod or is that not allowed? That would reduce the need for a dependency, or alternatively, the Unit can open up the Wireless Crafting Grid GUI when it detects that Refined Storage Addons is installed and then fall back to the normal Wireless Grid GUI if it detects that only Refined Storage is installed?

Also, there's no need to apologize for taking a break. Have a great holiday.

**Comment by MorningSage** [comment]
> ... or alternatively, the Unit can open up the Wireless Crafting Grid GUI when it detects that Refined Storage Addons is installed and then fall back to the normal Wireless Grid GUI if it detects that only Refined Storage is installed?

This was the route I was originally planning in my head.  This way, I could easily add support for other mods as they crop up.

One thought I've had is to make the unit in the mod currently into a "base" unit which can be combined with the specific grid/terminal you'd like in a crafting grid.  If possible, we could look up the terminal in either mod's registry (?) to determine the correct screen and use that.  If there's more than one grid/terminal, we could possibly have a radial selection like the Configurator has to select the one you want to open.  Thus, giving support for players to craft with gas, item, fluid, crafting, etc. terminals.

If the code is generic enough, it would support being crafted with any valid grid/terminal without explicitly adding support.  Again, this is just a thought and without a computer I can't check on the feasibility of implementing it.

Thoughts or concerns with that if it's possible?  There will likely be changes to the idea, but that would be the gist of it.
